components/org.wso2.bfsi.identity.extensions/src/main/java/org/wso2/bfsi/identity/extensions/auth/extensions/response/handler/impl/BFSIDefaultResponseTypeHandlerImpl.java

```diff
@@ -0,0 +1,171 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.bfsi.identity.extensions.auth.extensions.response.handler.impl;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.bfsi.consent.management.common.util.ConsentManagementConstants;
+import org.wso2.bfsi.identity.extensions.auth.extensions.response.handler.BFSIResponseTypeHandler;
+import org.wso2.bfsi.identity.extensions.internal.IdentityExtensionsDataHolder;
+import org.wso2.carbon.identity.oauth2.RequestObjectException;
+import org.wso2.carbon.identity.oauth2.authz.OAuthAuthzReqMessageContext;
+import org.wso2.carbon.identity.openidconnect.RequestObjectService;
+import org.wso2.carbon.identity.openidconnect.model.RequestedClaim;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Default extension implementation. Used to do the accelerator testing. Mimics a UK flow.
+ */
+public class BFSIDefaultResponseTypeHandlerImpl implements BFSIResponseTypeHandler {
+
+    private static final String OPENBANKING_INTENT_ID = "openbanking_intent_id";
+    private static final Log log = LogFactory.getLog(BFSIDefaultResponseTypeHandlerImpl.class);
+
+    /**
+     * return the new refresh validity period.
+     *
+     * @param oAuthAuthzReqMessageContext
+     * @return
+     */
+    public long updateRefreshTokenValidityPeriod(OAuthAuthzReqMessageContext oAuthAuthzReqMessageContext) {
+
+        return oAuthAuthzReqMessageContext.getRefreshTokenvalidityPeriod();
+    }
+
+    /**
+     * return the new approved scope.
+     *
+     * @param oAuthAuthzReqMessageContext
+     * @return
+     */
+    public String[] updateApprovedScopes(OAuthAuthzReqMessageContext oAuthAuthzReqMessageContext) {
+
+        if (oAuthAuthzReqMessageContext != null && oAuthAuthzReqMessageContext.getAuthorizationReqDTO() != null) {
+
+            String[] scopes = oAuthAuthzReqMessageContext.getApprovedScope();
+            if (scopes != null && !Arrays.asList(scopes).contains("api_store")) {
+
+                String sessionDataKey = oAuthAuthzReqMessageContext.getAuthorizationReqDTO().getSessionDataKey();
+                String consentID = getConsentIDFromSessionData(sessionDataKey);
+                if (consentID.isEmpty()) {
+                    log.error("Consent-ID retrieved from request object claims is empty");
```

Comment:
Shouldn't this be a warn log?

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/6/commits/884f24f4261bc92a886a68a4e8a5bf56e7a61136